Add initial how-to doc for planning app deployment

chenette opened the pull request:
Signed-off-by: Anne Chenette <chenette@bitwise.io>

davececchi commented:
This is a great document.  I'm not sure I see it as a "how-to" - it's more like "Deployment Patterns" in a "Getting Started" section or something.

I could easily see handing this to partners we work with.  Great stuff.

peterschwarz commented on docs/howto/planning_splinter_deployment.md:
```diff
@@ -0,0 +1,157 @@
+
+* **8044**: Network endpoint for node-to-node communication
+
+* **8085**: Application communication (encrypted with SSL) via the splinterd
```
This is not actually true.  The REST API for splinterd is served over standard HTTP.  If the deployment requires HTTPS, the current thinking is that there should be a proxy server for the HTTPS requests.

chenette replied:
Fixed.

chenette commented:
@davececchi - Thanks for the positive feedback!  The "category" for this topic is fluid; it's in `how-to` for now because it's not a concept topic, but can change as more information is developed.

chenette commented:
FYI, I pushed several fixup commits to add reviewer changes and make images consistent. Key changes:
- Removed "(encrypted with SSL)" from the port 8084 (line 74)
- Added a Summary section to wrap up the document (initial information for now; will expand in later PRs)
